src/python/pants/backend/makeself/subsystem.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import logging
+
+from pants.backend.makeself.goals.run import RunMakeselfArchive
+from pants.core.util_rules import external_tool
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolRequest,
+    TemplatedExternalTool,
+)
+from pants.engine.fs import Digest, RemovePrefix
+from pants.engine.platform import Platform
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+class MakeselfSubsystem(TemplatedExternalTool):
+    options_scope = "makeself"
+    help = "A tool to generate a self-extractable compressed tar archives."
+
+    default_version = "2.5.0"
+    default_known_versions = [
+        "2.5.0|macos_arm64 |4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+        "2.5.0|macos_x86_64|4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+        "2.5.0|linux_arm64 |4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+        "2.5.0|linux_x86_64|4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+    ]
```

Comment:
No particular need to change, but given this is the same for all platforms, feel free to reduce the repetition, similar to what PEX does: https://github.com/pantsbuild/pants/blob/db14a1095d6b0ed4a36b0bf84ee2cbd07085da07/src/python/pants/backend/python/util_rules/pex_cli.py#L42-L54